Makefile

```diff
@@ -22,13 +12,24 @@ ifneq ($(.SHELLSTATUS),0)
 endif
 
 ifeq ($(BRANCH),develop)
-  DOCKER_IMAGE_REPO := go-spacemesh
-else
   DOCKER_IMAGE_REPO := go-spacemesh-dev
+else
+  DOCKER_IMAGE_REPO := go-spacemesh
 endif
```

Comment:
Why change this?

Reply:
From the PR description:

```
For some reason builds on develop were pushed to go-spacemesh and on other branches (and releases) to go-spacemesh-dev. Changed to be the other way round
```

Reply:
cc @dshulyak

Reply:
I think it makes the most sense to always push to `go-spacemesh-dev` unless explicitly instructed to push to `go-spacemesh` (upon a new release).

I updated the PR to match this.